### DISCUSSION TITLE
get querystring data in javascript

### DISCUSSION THREAD

**Discussion by guidoDimasi** [elsa-workflows/elsa-core#5424] [Q&A]
assuming that i've a HttpRequest variable. how can i read the query string dictionary?
can you make an example in C# , Javascript and Liquid?

**Comment by glime-ai** [comment]
To read the query string dictionary from an `HttpRequest` variable, you can use different syntax depending on the language you are working with. Below are examples in C#, JavaScript, and Liquid:

**C# Example:**
```csharp
// Assuming you have an HttpRequest object named httpRequest
var queryStringValues = httpRequest.Query;

// To access a specific query string value
var specificValue = queryStringValues["key"]; // Replace 'key' with the actual key you want to retrieve
```

**JavaScript Example:**
```javascript
// Assuming you have a function getHttpRequest() that returns the HttpRequest object
var httpRequest = getHttpRequest();

// To access the query string dictionary
var queryStringDictionary = httpRequest.query;

// To access a specific query string value
var specificValue = queryStringDictionary["key"]; // Replace 'key' with the actual key you want to retrieve
```

**Liquid Example:**
```liquid
{% assign queryStrings = Request.Query %}

<!-- To iterate over all query string values -->
{% for queryParam in queryStrings %}
  {{ queryParam[0] }}: {{ queryParam[1] }}
{% endfor %}

<!-- To access a specific query string value -->
{{ queryStrings["key"] }} <!-- Replace 'key' with the actual key you want to retrieve -->
```

Please note that the JavaScript and Liquid examples assume that you have access to an `HttpRequest` object within your workflow context and that the object's properties are accessible using the provided syntax. The actual implementation details may vary depending on the workflow system you are using.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by guidoDimasi** [comment]
when writing the c# script inside a workflow (using workflow studio) i've this exception:

"error CS0234: Il tipo o il nome dello spazio dei nomi 'AspNetCore' non esiste nello spazio dei nomi 'Microsoft'. Probabilmente manca un riferimento all'assembly"
